testcases/OsSecureBoot.py

```diff
@@ -0,0 +1,169 @@
+        self.cv_BMC.run_command("echo '0 0x283a 0x15000000' > /var/lib/obmc/cfam_overrides")
+        self.cv_BMC.run_command("echo '0 0x283F 0x20000000' >> /var/lib/obmc/cfam_overrides")
+
+        self.cv_IPMI.ipmitool.run("raw 0x04 0x30 0xE8 0x00 0x40 0x00 0x00 0x00 0x00 0x00 0x00 0x00")
+        output = self.cv_IPMI.ipmitool.run("raw 0x04 0x2D 0xE8")
+        self.assertTrue("40 40 00 00" in output)
+        
+        self.cv_SYSTEM.sys_power_on()
+
+        raw_pty = self.cv_SYSTEM.console.get_console()
+
+        raw_pty.expect("Opened Physical Presence Detection Window", timeout=120)
+        raw_pty.expect("System Will Power Off and Wait For Manual Power On", timeout=30)
+        raw_pty.expect("shutdown complete", timeout=30)
+
+        # Shut itself off, turn it back on
+        # Need to turn it on by the BMC for some reason?
+        self.cv_BMC.run_command("obmcutil power on")
+
+        # This is apparently needed because otherwise op-test can't determine
+        # the state of the machine?
+        self.cv_SYSTEM.sys_check_host_status()
+
+        self.checkSecureBootEnabled(enabled=False, physical=True)
+        self.checkKeysEnrolled(enrolled=False)
+
+
+    def addSecureBootKeys(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        self.getTestData()
+
+        for k in ["PK", "KEK", "db", "dbx"]:
+            con.run_command("cat /tmp/{0}.auth > /sys/firmware/secvar/vars/{0}/update".format(k))
+
+        # System needs to power fully off to process keys on next reboot
+        # need to stall because otherwise it just blows on through on its own
+        # hopefully one of these two actually powers it off
+        self.cv_SYSTEM.sys_power_off()
+        time.sleep(10)
+        self.cv_SYSTEM.goto_state(OpSystemState.OFF)  
+        time.sleep(10)
+
+        self.checkSecureBootEnabled(enabled=True)
+        self.checkKeysEnrolled(enrolled=True)
+
+
+    def checkKexecKernels(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.PETITBOOT_SHELL)
+        con = self.cv_SYSTEM.console
+
+        self.getTestData(data="kernels")
+
+        # Fail regular kexec_load
+        output = con.run_command_ignore_fail("kexec -l /tmp/kernel-unsigned")
+        self.assertTrue("Permission denied" in "".join(output))
+
+        # Fail unsigned kernel
+        output = con.run_command_ignore_fail("kexec -s /tmp/kernel-unsigned")
+        self.assertTrue("Permission denied" in "".join(output))
+
+        # Fail dbx kernel
+        output = con.run_command_ignore_fail("kexec -s /tmp/kernel-dbx")
+        self.assertTrue("Permission denied" in "".join(output))
+        
+        # Succeed good kernel
+        output = con.run_command("kexec -s /tmp/kernel-signed")
+        self.assertFalse("Permission denied" in "".join(output)) # may not be needed, command should fail
+
+
+    def runTest(self):
+        # start clean
+        self.assertPhysicalPresence()
```

Comment:
It would be good to `checkSecureBootEnabled()`  and `skipTest()` otherwise before anything else.. Also, you can use goto_state(), and try to avoid additional reboot cycles...